### ISSUE TITLE
Warnings

### ISSUE THREAD

**Issue by garyb** [purescript/purescript-maps#40]
Compiling maps gives some pretty heinous pattern warnings, but I'm not sure the best way to work around them really... `throwUnsafe "The impossible happened"`?

```
Multiple warnings found:
  Warning in module Data.Map:
  Warning in value declaration delete:
  Warning at D:\Home\GitHub\purescript-maps\src\Data\Map.purs line 170, column 1 - line 171, column 1:
      Pattern could not be determined to cover all cases.
      The definition has the following uncovered cases:

      (Data.List.Cons (Data.Map.TwoRight Data.Map.Leaf _ _) _)                             (Data.Map.Two _ _ _ _)
      (Data.List.Cons (Data.Map.TwoRight Data.Map.Leaf _ _) _)                             (Data.Map.Three _ _ _ _ _ _ _)
      (Data.List.Cons (Data.Map.ThreeLeft _ _ Data.Map.Leaf _ _ (Data.Map.Two _ _ _ _)) _) _
      (Data.List.Cons (Data.Map.ThreeLeft _ _ Data.Map.Leaf _ _ (Data.Map.Two _ _ _ _)) _) (Data.Map.Two _ _ _ _)
      (Data.List.Cons (Data.Map.ThreeLeft _ _ Data.Map.Leaf _ _ (Data.Map.Two _ _ _ _)) _) (Data.Map.Three _ _ _ _ _ _ _)
      ...
  See https://github.com/purescript/purescript/wiki/Error-Code-NotExhaustivePattern for more information, or to contribute content related to this error.

  Warning in module Data.Map:
  Warning in value declaration delete:
  Warning at D:\Home\GitHub\purescript-maps\src\Data\Map.purs line 170, column 1 - line 171, column 1:
      Pattern could not be determined to cover all cases.
      The definition has the following uncovered cases:

      _ Data.Map.Leaf
  See https://github.com/purescript/purescript/wiki/Error-Code-NotExhaustivePattern for more information, or to contribute content related to this error.

  Warning in module Data.Map:
  Warning in value declaration delete:
  Warning at D:\Home\GitHub\purescript-maps\src\Data\Map.purs line 170, column 1 - line 171, column 1:
      Pattern could not be determined to cover all cases.
      The definition has the following uncovered cases:

      Data.Map.Leaf
  See https://github.com/purescript/purescript/wiki/Error-Code-NotExhaustivePattern for more information, or to contribute content related to this error.
```

**Comment by paf31** [comment]
Yeah, that's probably the best option.

**Comment by bagl** [comment]
Can be closed: 1b3fddaccc4e3b2e1dc3bbabdf7c60330fe9f200
